src/transaction_builder.js

```diff
@@ -153,33 +172,252 @@ export class TransactionBuilder {
    * @see TimeoutInfinite
    */
   setTimeout(timeout) {
+    if (timeout < 0) {
+      throw new Error('timeout cannot be negative');
+    }
+
+    return this.setMaxTime(Math.floor(Date.now() / 1000) + timeout);
+  }
+
+  /**
+   * If you want to prepare a transaction which will become valid at some point in the future, you can set a minimum time precondition.
+   * Internally this will set the <code>minTime</code> precondition.
+   *
+   * @param {Date|number} dateOrEpochSeconds Either a js Date object, or a number of unix epoch seconds. Can't be negative.
+   * If the value is `0`, the transaction is valid immediately.
+   * @return {TransactionBuilder}
+   * @see TransactionPreconditions
+   */
+  setMinTime(dateOrEpochSeconds) {
+    // Force it to a date type
+    if (typeof dateOrEpochSeconds === 'number') {
+      dateOrEpochSeconds = new Date(dateOrEpochSeconds * 1000);
+    }
+
+    // Convert that date to the epoch seconds
+    const t = Math.floor(dateOrEpochSeconds / 1000);
+    if (this.timebounds !== null && this.timebounds.minTime > 0) {
+      throw new Error(
+        'TimeBounds.min_time has been already set - setting min_time would overwrite it.'
+      );
+    }
+
+    if (t < 0) {
+      throw new Error('min_time cannot be negative');
+    }
+
+    if (this.timebounds === null) {
+      this.timebounds = {
+        minTime: 0,
+        maxTime: 0
```

Comment:
maybe we use `maxTime: TIMEOUT_INFINITE` here to self-document? but also I think this is breaking behavior: previously it was impossible to create a tx w/o timebounds unless you explicitly called `setTimeout(INF)` or `setTimeBounds(minTime: x, maxTime: INF)`, but now setting `setMinTime(x)` gives you inf max time _implicitly_ :thinking: 

No strong opinion, but what're your thoughts?

Reply:
I don't really see why creating a txn without a timeout is invalid to begin with. Maybe you're not going to submit it right away?

Reply:
We just discourage inf timeouts from being implicit behavior, so you must do `setTimeout(INF)`. I think the primary reason is because timebounds are the One True Way:tm: to get **definitive** success/failure state (see [Error Handling](https://developers.stellar.org/docs/tutorials/handling-errors/#retrying-until-success)), so we highly encourage people to set them no matter what.

Reply:
You could never set the min time before, so.. It's breaking either way. Maybe keeping `setTimeout(max)` and adding `setTimebounds(min,max)` would be a cleaner api. 🤔

Reply:
:100: agreed, much cleaner, or just `setTimeout(max)` and setting `timebounds` via the builder constructor (I thought setTimebounds existed for some reason).